Updates for ITKv5.2 code cleanups ... ROUND1

hjmjohnson opened the pull request.

hjmjohnson committed:
STYLE: Remove CMake-language block-end command arguments

Ancient versions of CMake required else(), endif(), and similar block
termination commands to have arguments matching the command starting the block.
This is no longer the preferred style.

hjmjohnson committed:
STYLE: Convert CMake-language commands to lower case

Ancient CMake versions required upper-case commands.  Later command names
became case-insensitive.  Now the preferred style is lower-case.

hjmjohnson committed:
STYLE: Prefer = default to explicitly trivial implementations

This check replaces default bodies of special member functions with
= default;. The explicitly defaulted function declarations enable more
opportunities in optimization, because the compiler might treat
explicitly defaulted functions as trivial.

Additionally, the C++11 use of = default more clearly expreses the
intent for the special member functions.

hjmjohnson committed:
STYLE: Remove redundant void argument lists

Find and remove redundant void argument lists.

hjmjohnson committed:
DOC: Update copyright assignment to NumFOCUS

The mission of NumFOCUS is to promote open
practices in research, data, and scientific
computing.

https://numfocus.org

dzenanz commented on include/itkVariationalRegistrationFastNCCFunction.hxx:
```diff
@@ -19,7 +19,7 @@
 
 #include "itkVariationalRegistrationFastNCCFunction.h"
-#include "itkExceptionObject.h"
+#include "itkMacro.h"
```
Misleading commit message

hjmjohnson committed:
ENH: Add .gitattributes to allow running ITK clang-formatting scripts

```
git filter-branch -f \
--tree-filter "~/ITK/Utilities/Maintenance/clang-format.bash --clang-format ~/Dashboard/src/ITK-clang11/clang-format-Linux --tracked" \
    master..
```

hjmjohnson committed:
BUG: Fix incorrect wrapping syntax for object types

The wrapping paradim for different object types requires
different wrapping functions to be use.

```
CMake Error at Wrapping/TypedefMacros.cmake:730 (message):
  Wrapping itk::VariationalRegistrationCurvatureRegularizer: No image type
  for '2+' pixels is known.
Call Stack (most recent call first):
  Wrapping/TypedefMacros.cmake:668 (itk_wrap_image_filter_types)
  Modules/Remote/VariationalRegistration/wrapping/itkVariationalRegistrationCurvatureRegularizer.wrap:2 (itk_wrap_image_filter_combinations)
  Wrapping/TypedefMacros.cmake:219 (include)
  Wrapping/TypedefMacros.cmake:173 (itk_load_submodule)
  Modules/Remote/VariationalRegistration/wrapping/CMakeLists.txt:21 (itk_auto_load_submodules)
```

hjmjohnson commented:
@jhlegarreta FYI: Please look at the last patch https://github.com/InsightSoftwareConsortium/ITKVariationalRegistration/pull/40/commits/39acd985bf86534d037a68a17f729791f70efe8e

hjmjohnson committed:
ENH: Adding azure pipeline script.

jhlegarreta commented:
@hjmjohnson Reading the error message your editions look fair, but would not be able to confirm the expected behavior.

I would remove the commented lines from the `wrapping` files. Not sure if you added them to give them a try.

The `Evaluate template` step failure the CI pipelines are reporting is somehow cryptic. Not sure what's the source of the issue.

hjmjohnson commented:
@jhlegarreta I used code that matched code in ITK wrapping for similar objects.  In all cases, ITK also had the commented out wrapping code too, so I kept it the same here.

I'm working on the CI, but will not try to make it work 100%.  I only have time to make it better than it was.

jhlegarreta commented:
OK, thanks for the explanation Hans.

Again :100: to the effort you are making on the remotes.
